lib/validation.js

```diff
@@ -16,6 +16,13 @@ module.exports = function validation (yargs, usage, y18n) {
     const demandedCommands = yargs.getDemandedCommands()
     // don't count currently executing commands
     const _s = argv._.length - yargs.getContext().commands.length
+    const commandKeys = yargs.getCommandInstance().getCommands()
+
+    // for the special case of yargs.demandCommand() and yargs.demandCommand(1), if
+    // yargs has been configured with commands, we automatically enable strictCommands.
+    if (commandKeys.length && demandedCommands._ && demandedCommands._.min === 1 && demandedCommands._.max === Infinity) {
+      yargs.strictCommands()
```

Comment:
If this is the only reason the PR is considered a breaking change, I would argue in favor of postponing this particular behavior until later, and allowing a normal feature release for this. I'm not convinced the major version fatigue is worth it, here.

Reply:
@evocateur the new version of [yargs-parser](https://github.com/yargs/yargs-parser/blob/master/CHANGELOG.md#-breaking-changes) has two, I think important, breaking changes that I'd like to shield us from with a major in `yargs`.

Reply:
Ah, fair. Might as well, then.